### PR TITLE
fix flakey get namespace unit test

### PR DESCRIPTION
**Pull request by nader-ziada**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
- this unit test fails sometimes when run in parallerl since it reads from the os environment variable 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: util/system/namespace_test.go [hunk 1]
```diff
@@ -44,10 +44,10 @@ func TestGetNamespace(t *testing.T) {
 	for _, c := range cases {
 		c := c
 		t.Run(c.Name, func(t *testing.T) {
-			t.Parallel()
 			g := gomega.NewWithT(t)
 			os.Setenv(NamespaceEnvVarName, c.PodNamespace)
 			g.Expect(GetManagerNamespace()).To(gomega.Equal(c.Expected))
+			os.Unsetenv(NamespaceEnvVarName)
```
[the remark this answers is not in the document]
Comment (nader-ziada) [reply util/system/namespace_test.go]: done

File: util/system/namespace_test.go [hunk 2]
```diff
@@ -44,10 +44,11 @@ func TestGetNamespace(t *testing.T) {
 	for _, c := range cases {
 		c := c
 		t.Run(c.Name, func(t *testing.T) {
-			t.Parallel()
 			g := gomega.NewWithT(t)
 			os.Setenv(NamespaceEnvVarName, c.PodNamespace)
+			defer os.Unsetenv(NamespaceEnvVarName)
 			g.Expect(GetManagerNamespace()).To(gomega.Equal(c.Expected))
+
```
Comment (devigned) [inline util/system/namespace_test.go:51]: super nit; please remove the newline

**Review by devigned — CHANGES_REQUESTED** [review]

File: util/system/namespace_test.go (hunk 2 above)
Comment (nader-ziada) [reply util/system/namespace_test.go]: done

**Pull request merged**